### PR TITLE
Beta version graphql

### PR DESCRIPTION
**Pull request by denisart**
Hello, thank for your project. Code generation it is an actual problem and `datamodel-code-generator` it is very useful tool.

I created code generators:
- [graphql-query](https://github.com/denisart/graphql-query): Complete Domain Specific Language (DSL) for GraphQL query in Python. (Added to [graphql.org](https://graphql.org/code/#python))
- [graphql2python](https://github.com/denisart/graphql2python): A tool that generates python code out of your GraphQL schema.

My `graphql2python` project is still raw. But `datamodel-code-generator` has a nice core for generation code from a graphql schema. I created a beta version of graphql parser based on `datamodel_code_generator.parser.base.Parser`.

So far I have added only one test to present the work of the parser. If this idea is interesting for this project, then I am ready to develop and support functionality for generating code from a graphql schema.

### CODE REVIEW

**Comment by koxudaxi** [comment]
@denisart 
Thank you for creating this PR!!
I'm not familiar with GraphQL. but I guess the generated pydantic model is useful.
In order to get more helpful feedback, we want to incorporate features that everyone uses more frequently.
I think the feature matches that.

Could you please add documentation and CLI argument?

**Comment by denisart** [comment]
@koxudaxi
Thanks for answer.

I suggest to add docs (for graphql case) and a few tests (for openapi like cases i think) to this PR which showing some usage examples.

What do you mean by CLI argument?

**Comment by koxudaxi** [comment]
@denisart 
> I suggest to add docs (for graphql case) and a few tests (for openapi like cases i think) to this PR which showing some usage examples.

I agree.
Could you please add it?


> What do you mean by CLI argument?

 `GraphQLScope`
 My mistake.
I thought this enum was something that could be selected from the CLI, but it is currently one type.
please ignore my question.

**Comment by denisart** [comment]
@koxudaxi 

I added docs for graphql and a few tests.
I can test the resulting version in my work, where I use GraphQL in Python.

File: datamodel_code_generator/model/template/Union.jinja2 [hunk 1]
```diff
@@ -0,0 +1,10 @@
+{%- if description %}
+# {{ description }}
+{%- endif %}
+{%- if fields|length > 1 %}
+{{ class_name }}: TypeAlias = Union[
```
Comment (koxudaxi) [inline datamodel_code_generator/model/template/Union.jinja2:5]: `TypeAlias`  is good for defining types.
We would apply `TypeAlias` to other output models like TypedDict and dataclasses next time.

**Review by koxudaxi — APPROVED** [review]

**Pull request merged**